### DISCUSSION TITLE
Feature request: ini.managed all-in-one state

### DISCUSSION THREAD

**Discussion by oeuftete** [saltstack/salt#67489] [Feature Requests]
### Description of Issue

(ref. ZD-4366)

To simplify the maintenance and change reporting of ini file management states, especially when building an ini from a standard starting point, it would be useful to have an all-in-one state, perhaps `ini.managed`.

Here's how it might look.

```sls
all_in_one_ini_state:
  ini.managed:
    - name: target.ini
    - source: salt://vanilla.ini
    - options:
        present:
          test:
            testkey: 'testval'
        absent:
          test:
            - badkey
    - sections:
        present:
          - good_empty_section
        absent:
          - bad_section
```

That state would get the source file, perform the various options and sections modifications to it as in the [standalone ini states](https://docs.saltstack.com/en/latest/ref/states/all/salt.states.ini_manage.html), and only if that result differs from `target.ini` would the state return with changes.

The current workaround looks a bit like:

```sls
{% set SALT_REFERENCE = 'salt://vanilla.ini' %}
{% set LOCAL_REFERENCE = 'local_reference.ini' %}
{% set LOCAL_TMP = '/tmp/tmp.ini' %}
{% set LOCAL_FINAL = 'target.ini' %}

get_ini_from_source_if_changed:
  file.managed:
    - name: {{ LOCAL_REFERENCE }}
    - source: {{ SALT_REFERENCE }}

rebuild_ini_if_reference_changed:
  file.managed:
    - name: {{ LOCAL_TMP }}
    - source: {{ SALT_REFERENCE }}
    - onchanges:
      - file: get_ini_from_source_if_changed

apply_ini_modifications:
  ini.options_present:
    - name: {{ LOCAL_TMP }}
    - sections:
        test:
          testkey: 'testval'

{# ... more ini states possibly ... #}

copy_tmp_ini_to_target:
  file.managed:
    - name: {{ LOCAL_FINAL }}
    - source: {{ LOCAL_TMP }}
```

**Comment by flassman** [comment]
The proposed feature would be exactly what I need. The workaround has the drawback that if I remove one of the ini modifications from the state and run again, change is not reflected in the target until the source changes and the target is recreated.
